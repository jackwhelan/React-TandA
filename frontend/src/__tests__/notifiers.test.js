import React from 'react';
import renderer from 'react-test-renderer';

import Alert from '../components/notifiers/Alert.component';
import Greeting from '../components/notifiers/Greeting.component';
import Info from '../components/notifiers/Info.component';

describe('Notifier Components', () => {
    test('Alert', () => {
        const component = renderer.create(<Alert />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Greeting', () => {
        const component = renderer.create(<Greeting />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Info => Error', () => {
        const component = renderer.create(<Info status="error" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Info => Warning', () => {
        const component = renderer.create(<Info status="warning" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Info => Success', () => {
        const component = renderer.create(<Info status="success" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})