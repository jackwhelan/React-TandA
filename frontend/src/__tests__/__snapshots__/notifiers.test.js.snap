// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifier Components Alert 1`] = `
<div
  className="mt-2 text-danger"
/>
`;

exports[`Notifier Components Greeting 1`] = `
<div
  className="border border-dark rounded p-4"
>
  <h6>
    Hello null
  </h6>
  <hr
    className="bg-dark"
  />
</div>
`;

exports[`Notifier Components Info => Error 1`] = `
<div
  className="infoComponent"
  style={
    Object {
      "backgroundColor": "#d9534f",
    }
  }
>
  <h5 />
  <p />
</div>
`;

exports[`Notifier Components Info => Success 1`] = `
<div
  className="infoComponent"
  style={
    Object {
      "backgroundColor": "#5cb85c",
    }
  }
>
  <h5 />
  <p />
</div>
`;

exports[`Notifier Components Info => Warning 1`] = `
<div
  className="infoComponent"
  style={
    Object {
      "backgroundColor": "#d9534f",
    }
  }
>
  <h5 />
  <p />
</div>
`;
